scripts/db_backup.sh

```diff
@@ -242,18 +255,58 @@ copy_db_to_azure_storage()
 
 cleanup_local_copies()
 {
-    pushd $DESTINATION_FOLDER
+    pushd $BACKUP_LOCAL_PATH
 
     log "Deleting local copies of $DATABASE_TYPE database"
     rm -rf $COMPRESSED_FILE
     rm -rf $BACKUP_PATH
 
+    # And any other backups.
+    rm -rf *${CONTAINER_NAME}*
+
     rm -rf $TMP_QUERY_ADD
     rm -rf $TMP_QUERY_REMOVE
 
     popd
 }
 
+cleanup_old_remote_files()
+{
+    cutoffInSeconds=$1
+
+    # This is very noisy. We'll use log to communicate status.
+    set +x
+
+    log "Getting list of files and extracting their age"
+    log "files older than $BACKUP_RETENTIONDAYS days will be removed"
+
+    # Get file list with lots of meta-data. We'll use this to extract dates.
+    verboseDetails=`azure storage blob list $CONTAINER_NAME --json`
+
+    # List of files (generally formatted like this: mysql-backup_2017-04-20_03h-00m-01s.tar.gz)
+    terminater="\"" # quote
+    fileNames=`echo "$verboseDetails" | jq 'map(.name)' | grep -oE "$terminater.*$terminater" | tr -d $terminater`
+    # FYI, another approach is to use the file's timestamp which /bin/date can handle natively
+    #fileStamp=`echo "$verboseDetails" | jq 'map(.lastModified)'`
+
+    while read fileName; do
+        # Parse time from file. Something like  2017-04-20_03h-00m-01s
+        #   and convert it to somethign like    2017-04-20 03:00:01
+        terminater="s\." # s.
+        fileDateString=`echo "$fileName" | grep -o "[0-9].*$terminater" | sed "s/$terminater//g" | sed "s/h-\|m-/:/g" | tr '_' ' '`
+        fileDateInSeconds=`date --date="$fileDateString" +%s`
+
+        if [ $cutoffInSeconds -ge $fileDateInSeconds ]; then
+            log "deleting $fileName"
+            azure storage blob delete -q $CONTAINER_NAME $fileName
+        else
+            log "keeping $fileName"
+        fi
+    done <<< "$fileNames"
```

Comment:
the short answer:
< reads input from console (command line terminal),
<< reads input from a document (generally from file system disk)
<<< reads input from a string (often a bash variable)

Some References
1 "Bash includes the ‘<<<’ redirection operator, allowing a string to be used as the standard input to a command." from https://www.gnu.org/software/bash/manual/bashref.html
2 [a string] "is expanded and fed to the stdin of command" from description of "here strings" on https://linux.die.net/abs-guide/x15683.html and example uses of the "triple-less-than" operator
3 also covered on https://en.wikipedia.org/wiki/Less-than_sign#Triple_less-than_sign